src/test/regress/sql/with.sql

```diff
@@ -1065,3 +1065,39 @@ WITH cte1 AS (
 SELECT * FROM cte1 a JOIN cte1 b USING (c1);
 
 DROP TABLE t1;
+
+-- Test join prefetch_inner in the case of bottleneck join
+SET optimizer = off;
+--start_ignore
+DROP TABLE IF EXISTS d;
+--end_ignore
+CREATE TABLE d (c1 int, c2 int) DISTRIBUTED BY (c1);
+
+INSERT INTO d (VALUES ( 2, 0 ),( 2 , 0 ));
+
+-- Test with HashJoin
+EXPLAIN (ANALYZE, COSTS OFF) WITH cte AS (
+	SELECT count(*) c1 FROM d
+) SELECT * FROM cte a JOIN (SELECT * FROM d JOIN cte USING (c1) LIMIT 1) b USING (c1);
+
+SET enable_hashjoin = off;
+SET enable_nestloop = on;
+
+-- Test with nestedloop join
+EXPLAIN (ANALYZE, COSTS OFF) WITH cte AS (
+	SELECT count(*) c1 FROM d
+) SELECT * FROM cte a JOIN (SELECT * FROM d JOIN cte USING (c1) LIMIT 1) b USING (c1);
+
+SET enable_nestloop = off;
+SET enable_mergejoin = on;
+
+-- Test with merge join
```

Comment:
same question: 

What exactly is this case testing if it produces the same result even without a patch?

Reply:
yes, the behavior does not change, but it may make sense to leave them since the original PR(https://github.com/arenadata/gpdb/commit/b583950a855c6d1fa37a79600233c6b65df648a8) does not cover this logic with tests?

Reply:
Then maybe add a comment that this is a check that the patch didn't break anything, what worked before?